service/history/workflow/workflow_task_state_machine.go

```diff
@@ -789,14 +743,13 @@ func (m *workflowTaskStateMachine) tryRestoreSpeculativeWorkflowTask(
 
 func (m *workflowTaskStateMachine) setSpeculativeWorkflowTaskStartedEventID(
 	workflowTask *WorkflowTaskInfo,
-	startedEventID int64,
 ) {
 	// TODO (alex-update): Uncomment this code to support speculative workflow task restoration.
 
 	/*
 		// StartedEventID might be lost (cleared) for speculative workflow task due to shard reload or history service restart.
 		if workflowTask != nil && workflowTask.Type == enumsspb.WORKFLOW_TASK_TYPE_SPECULATIVE && workflowTask.StartedEventID == common.EmptyEventID {
-			workflowTask.StartedEventID = startedEventID
+			workflowTask.StartedEventID = workflowTask.ScheduledEventID + 1
```

Comment:
Started ID is always scheduled ID + 1? I don't recall we buffer events when there's a scheduled workflow task, or is speculative workflow task different?

Reply:
For normal WT we don't buffer events after scheduled WT, we only buffer them after WT is started. This formula won't work for it. Speculative (and transient) WT gets converted to normal when started, if there are new events came after it was scheduled.